tests/e2e.sh

```diff
@@ -100,6 +100,35 @@ test_cluster_simultaneous_start() {
     stop_test_container test1-orig
 }
 
+test_multiple_clusters_restart() {
+    cluster1='--cluster-port 7946 --wireguard-port 51820 --interface wgoverlay --overlay-net 10.10.0.0/16'
+    cluster2='--cluster-port 7947 --wireguard-port 51821 --interface wgoverlay2 --overlay-net 10.11.0.0/16'
+    setup_wireguard='wireguard-go wgoverlay2 2>/dev/null >/dev/null'
+    join_cluster2="nohup /app/wesher --cluster-key 'ILICZ3yBMCGAWNIq5Pn0bewBVimW3Q2yRVJ/Be+b1Uc=' --join test2-orig $cluster2 2>/dev/null &"
+    
+    run_test_container test1-orig test1 --init $cluster1
+    run_test_container test2-orig test2 --init $cluster2
+    run_test_container test3-orig test3 --join test1-orig $cluster1
+    docker exec test3-orig bash -c "$setup_wireguard; $join_cluster2"
+    
+    sleep 3
+
+    docker stop test3-orig
+    docker start test3-orig
+    docker exec test3-orig bash -c "$setup_wireguard; $join_cluster2"
```

Comment:
We probably want to restart it _without_  the explicit `--join` flag, right? To test the loading of the state file?

Now that I mention it, `test_node_restart` is also not testing what I wanted it to test. I should probably fix that :unamused:

Reply:
Done in the latest commit :)